### PR TITLE
bluetooth: mesh: CTL Temperature binding fix

### PR DESCRIPTION
**Pull request by omkar3141**
CTL temperature bindings should use rounding
operation for division in the binding formula.

Signed-off-by: Omkar Kulkarni <omkar.kulkarni@nordicsemi.no>

### CODE REVIEW

**Review by joerchan — APPROVED** [review]
Is the dependent fix included in the current upmerge?

**Comment by omkar3141** [comment]
> Is the dependent fix included in the current upmerge?

There is no dependency now.

File: subsys/bluetooth/mesh/light_ctl_internal.h [hunk 1]
```diff
@@ -17,19 +17,20 @@
 static inline uint16_t lvl_to_temp(struct bt_mesh_light_temp_srv *srv,
 				   int16_t lvl)
 {
-	return srv->range.min + (lvl + ((UINT16_MAX / 2) + 1)) *
-					(srv->range.max - srv->range.min) /
-					UINT16_MAX;
+	return srv->range.min +
+	       ROUNDED_DIV((int32_t)(lvl + ((UINT16_MAX / 2) + 1)) *
+				(srv->range.max - srv->range.min),
+				(int32_t)UINT16_MAX);
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/light_ctl_internal.h:23]: This is looking very busy. Consider putting `((UINT16_MAX / 2) + 1)` in a define, such as `LIGHT_TEMP_LVL_OFFSET`. 

Also, its definition seems a bit weird. I think setting it to `0 - BT_MESH_LVL_MIN` makes more sense, it's accounting for the offset between LVL_MIN and LIGHT_TEMP_MIN, which is `BT_MESH_LVL_MIN` and `0`, respectively. Or flip it around, and make it `BT_MESH_LVL_MIN`.

Could also put `srv->range.max - srv->range.min` in a variable

Comment (omkar3141) [reply subsys/bluetooth/mesh/light_ctl_internal.h:23]: I tried to keep it as it was. Considering your comment, I think I will just use the formula as in specification with actual numbers, which keeps it simple and easy to follow.

Comment (trond-snekvik) [reply subsys/bluetooth/mesh/light_ctl_internal.h]: This seems like a degrade to me, tbh. `UINT16_MAX` communicated that the number was supposed to be the highest representable number, which is technically the intent in the spec as well

File: subsys/bluetooth/mesh/light_ctl_internal.h [hunk 2]
```diff
@@ -14,22 +14,25 @@
 #include <bluetooth/mesh/light_temp_srv.h>
 #include "model_utils.h"
 
+#define LIGHT_TEMP_LVL_OFFSET 32768
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/light_ctl_internal.h:17]: (-INT16_MIN)

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Pull request merged**